File: backend/src/main/java/ch/clicktotranslate/vocabulary/infrastructure/web/EntryDto.java
package ch.clicktotranslate.vocabulary.infrastructure.web;

import ch.clicktotranslate.vocabulary.domain.Language;
import java.time.Instant;
import java.util.List;

public record EntryDto(Long entryId, Language language, String term, String termCustomization,
		List<TermDto> translations, List<UsageDto> usages, Instant lastEdit, Instant createdAt) {
}

File: web-app/vocabulary/src/app/vocabulary/infrastructure/dto/entry.dto.ts
import { TermDto } from './term.dto';
import { UsageDto } from './usage.dto';

export interface EntryDto {
  entryId: number;
  language: string;
  term: string;
  termCustomization?: string;
  translations: TermDto[];
  usages: UsageDto[];
  lastEdit: string;
  createdAt: string;
}
